src/storage-client/src/source/persist_source.rs

```diff
@@ -295,3 +305,44 @@ impl PendingWork {
         true
     }
 }
+
+#[derive(Debug)]
+struct PersistSourceDataStatsImpl<'a> {
+    stats: &'a PartStats,
+}
+
+impl PersistSourceDataStats for PersistSourceDataStatsImpl<'_> {
+    fn len(&self) -> Option<usize> {
+        None
+    }
+
+    fn err_count(&self) -> Option<usize> {
+        None
+    }
+
+    fn col_min<'a>(&self, name: &str, arena: &'a RowArena) -> Option<Datum<'a>> {
+        self.stats
+            .opt_u64_key_col_min_max_nulls(name)
+            .map(|(min, _, _)| arena.make_datum(|p| p.push(Datum::from(min))))
+    }
+
+    fn col_max<'a>(&self, name: &str, arena: &'a RowArena) -> Option<Datum<'a>> {
+        self.stats
+            .opt_u64_key_col_min_max_nulls(name)
+            .map(|(_, max, _)| arena.make_datum(|p| p.push(Datum::from(max))))
```

Comment:
This could be equivalently and more efficiently implemented as `Datum::Uint64(max)`, yeah?

(I assume that this is here for future-proofing and not because we really need it for `u64`... can you say a bit more about what that future looks like?)

Reply:
yeah, great question! for something like `Datum::String` or `Datum::Bytes`, we could put the lifetime bound on `&self` to simply wrap the alloc inside the persist stats object. we need the RowArena for e.g. `Datum::List`

now that I'm thinking about this again, we might be able to get the best of all worlds if both `&self` and `&RowArena` are tied to the output lifetime. let me give that a shot